### DISCUSSION TITLE
Changes to header and such

### DISCUSSION THREAD

**Discussion by veronica-yyc** [kudrykv/latex-yearly-planner#75] [Q&A] [answered]
Hi, I changed a few things to make this template perfect for work. But I'd love a little help.

1 - I changed labels in daily reflect to be a Meetings Notes Page.  On the daily planner page, it now says Meeting Minutes and it links to the old reflect page with new headings but I can't figure out how to change the Non Active hyperlink in the header to read "Meeting Minutes" instead of "Reflect".
2 - I'd love to remove the vertical lines between all the headers ( found the tabularx but no "|" to remove ) - not sure where they are.
3- Would love to add another notes section for Meeting Minutes for longer meetings with an index that has a spot for Project Name and Description/Date.  I think I can just copy and make a few changes but a list of all the files that would need revising - if someone is very familiar - would be awesome.
4 - Remove the boundaries around the day numbers in the monthly review page.
5- Possibly add a quarters page in 3 columns with the dates next to... thats a big change... not holding my breath. 
6 - Oh, I add my name to front page - not a bad touch if could be set as a user input i.e., bash build 2023 "MyNameIs"
Really love the template. 

Any tips would be great! Thank you.

**Comment by kudrykv** [comment, marked as answer]
Hi @veronica-yyc!

**P.1**: do you refer to this part?
![CleanShot 2023-03-01 at 09 04 19](https://user-images.githubusercontent.com/12697803/222067744-e172e283-19f5-48d9-a4cc-83c481b4f265.png)
It should be this line: https://github.com/kudrykv/latex-yearly-planner/blob/e4b39ece1e0d26cf7fed8ac27ef951f2f007ed3b/app/compose/daily.go#L10
The second parameter is what's being printed

**P.2**: try this: https://github.com/kudrykv/latex-yearly-planner/blob/e4b39ece1e0d26cf7fed8ac27ef951f2f007ed3b/app/components/header/items.go#L19-L25
The vertical `|` resembles the vertical line

**P.3**: So the code is tangled more than it should be. [In the app.go](https://github.com/kudrykv/latex-yearly-planner/blob/e4b39ece1e0d26cf7fed8ac27ef951f2f007ed3b/app/app.go#L134-L144) there is a list of "modules" that are available — to add new ones, you need to create one there. Then as an example you can look how they are implemented.

The hardest part is building the header; the next hard part is cross-linking between modules.

For example, [building daily](https://github.com/kudrykv/latex-yearly-planner/blob/e4b39ece1e0d26cf7fed8ac27ef951f2f007ed3b/app/compose/daily.go):
- we return `page.Modules` which is a slice. Number of returned modules should equal the number of rendered pages
- `cfg` is sort-of pass-through, just should be there. It's a global config, that you may pay attention to (left hand / right hand, AMPM / 2400, etc)
- `tpl` is the template to pick and to render
- `body` is the data that one can use in the template, and templates are located in `%projectRoot%/tpls`

Looking at [notes pages](https://github.com/kudrykv/latex-yearly-planner/blob/e4b39ece1e0d26cf7fed8ac27ef951f2f007ed3b/app/compose/notes_indexed.go#L10) you can see how there can be more than one template file


**P.4**: should be this part https://github.com/kudrykv/latex-yearly-planner/blob/e4b39ece1e0d26cf7fed8ac27ef951f2f007ed3b/app/components/cal/day.go#L27C6-L29

**P.5**: not sure I got that, can you please explain? Although, I was having a little energy to support this thing lately anyway 😞

**P.6**: that's a great idea for me to look into in the future!

**Reply by veronica-yyc** [reply]
I think you've given me enough to figure it out. 

I managed to make another notes section and relabel it but couldnt figure out how to hyperlink it with the header. However, I'll give it another go this weekend with your input. Thank you so much!
